apollo-router/src/plugins/telemetry/tracing/jaeger.rs

```diff
@@ -102,42 +97,21 @@ impl TracingConfigurator for Config {
                 tracing::info!("Configuring Jaeger tracing: {}", batch_processor);
                 // We are waiting for a release of https://github.com/open-telemetry/opentelemetry-rust/issues/894
                 // Until that time we need to wrap a tracer provider with Jeager in.
-                let tracer_provider = opentelemetry_jaeger::new_collector_pipeline()
+                let exporter = opentelemetry_jaeger::new_collector_pipeline()
                     .with_trace_config(trace_config.into())
                     .with_service_name(trace_config.service_name.clone())
                     .with(&collector.username, |b, u| b.with_username(u))
                     .with(&collector.password, |b, p| b.with_password(p))
                     .with_endpoint(&collector.endpoint.to_string())
                     .with_reqwest()
                     .with_batch_processor_config(batch_processor.clone().into())
-                    .build_batch(opentelemetry::runtime::Tokio)?;
-                Ok(builder
-                    .with_span_processor(DelegateSpanProcessor { tracer_provider }.filtered()))
+                    .build_collector_exporter::<runtime::Tokio>()?;
+                Ok(builder.with_span_processor(
+                    BatchSpanProcessor::builder(exporter, opentelemetry::runtime::Tokio)
+                        .with_batch_config(batch_processor.clone().into())
+                        .build(),
+                ))
             }
         }
     }
 }
-
```

Comment:
Jaeger no-longer needs this workaround as it is possible to create the exporter without setting it as the default tracer pipeline.